nilearn/maskers/surface_labels_masker.py

```diff
@@ -222,18 +221,9 @@ def fit(self, imgs=None, y=None):
         if imgs is not None:
             self._check_imgs(imgs)
 
-        if imgs is not None:
-            check_surf_img(imgs)
-
-            if isinstance(imgs, SurfaceImage) and any(
-                hemi.ndim > 2 for hemi in imgs.data.parts.values()
-            ):
-                raise ValueError(
-                    "should only be SurfaceImage should 1D or 2D."
-                )
```

Comment:
removing this in all maskers as we do not handled 3D surface image (with several 'layers') for now: added a check and a test for this in surface image creation